### DISCUSSION TITLE
Microsoft.StackExchange.Redis  does not let adding new connection to existing groups.

### DISCUSSION THREAD

**Discussion by panoj** [dotnet/aspnetcore#22750] [General]
src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

The AddToGroupAsync in the RedisHubLifeTimeManager does not allow adding new connections to existing group.

 Please let me know how do i navigate past this issue.

Below method is called from AddToGroupAsync.  The condition If(groups.Add(groupName)) does not allow to add connection to existing groups.

 private Task AddGroupAsyncCore(HubConnectionContext connection, string groupName)
        {
            var feature = connection.Features.Get<IRedisFeature>();
            var groupNames = feature.Groups;

            lock (groupNames)
            {
                // Connection already in group
                if (!groupNames.Add(groupName))
                {
                    return Task.CompletedTask;
                }
            }

            var groupChannel = _channels.Group(groupName);
            return _groups.AddSubscriptionAsync(groupChannel, connection, SubscribeToGroupAsync);
        }

t

**Comment by BrennanConroy** [comment]
Are you seeing an issue or just reading the code?

The code checks what groups the connection is already in and either adds it to the group or not depending on that.

**Comment by BrennanConroy** [comment]
Continuing at https://github.com/dotnet/aspnetcore/issues/22751
